### ISSUE TITLE
Re-enable Fortune for ore block drops, but make the drop behaviour configurable.

### ISSUE THREAD

**Issue by AncientShotgun** [GregTechCEu/GregTechCEu-1.19#1653]
### Feature Description

I was reading through the changelogs for your old releases, and came across the one where your team disabled Fortune-enchanted mining tools from multiplying raw ore drops. So I took a look at your MixinHelpers class file and noticed that the relevant section of it still hadn't changed from that old release.

I was thinking about how to solve this balancing issue, only to come upon the idea that every data-driven developer of FOSS products has at least once daily: Leave the choice up to the user!

Here's a more detailed implementation suggestion:
Make a new config entry in the common config file with a reference-able ID like 'oreFortuneBehavior' or similar. The config entry should be able to assume three states, determined by a string parameter the user can change:
'CLASSIC': Fortune does not affect the raw ore drop amount, multiplication is done in the raw ore maceration stage of ore processing, producing a number of the respective crushed ore items + byproducts according to the ore's oreMultiplier property. This causes ore processing behavior similar to GTCEu in MC versions like 1.12.x or 1.7.x.
'MODERN'/'DEFAULT': Similar to the CLASSIC setting in that Fortune plays no part in increasing the raw ore drop amount, but the oreMultiplier property defines the number of raw ore items dropped when the ore block is mined and each raw ore item only macerates to one (1) of its respective crushed ore item + byproducts. This is intended to be the default setting, and also the fallback setting in case the user writes text that isn't any valid setting in the string field for this config entry. If I read your codebase correctly, believe this is also how raw ore drops function now, in the 1.3.x versions of GTCEu Modern; please correct me if I'm wrong.
'VANILLA': For this setting, both the oreMultiplier property and the level of Fortune the user's mining tool has at time of mining the ore block influences the amount of raw ore dropped. Like in the 'MODERN' setting, one (1) raw ore macerates to one (1) crushed ore + byproducts. This is similar to how vanilla copper, redstone and lapis lazuli behave. Very overpowered when mods like Apotheosis (that jack up the max level of Fortune a player can achieve during regular gameplay without commands) are loaded alongside GTCEu.

The content of this suggestion so far is operating based on the assumption that an ore Material's oreMultiplier property accounts for both if the ore block is a dimensional variant (nether and end ores drop 2x the ore, as you know) and if the ore is designed to be 'crumbly', like vanilla copper, redstone or lapis lazuli, and drop multiple ore items even without Fortune on the player's mining tool. Please correct me on this assumption if I'm wrong.

That's the end of this initial suggestion. I am open to improvements and critique if you or your team have any.

**Comment by Ghostipedia** [comment]
GTM Forces ores to not work with fortune for a few reasons.

1 - Ore Duplication exists, and veins are already overturned in resource yields from the days of 1.12 CEU.
2 - Allowing this configuration would throw any semblance of balance out the window for the whole mod. While I would be for this change, it has too large of an impact to consider against the stock progression. You also have to consider that mods that allow greater than Fortune 3 Enchanting would completely annihilate balancing and viability of chunk miners as it would become more viable to mine with a drill stacked with fortune 10 than it would be to slap down a chunk miner and ore-proc. 
3 - If something like this was implemented, it would not happen on the raw ore step, I would think it would make more sense to give more bonus rolls or boost the byproduct chances as you mine VS just "more ore chunks"
4 - your  second option here for 'modern/default' - is based on an old issue where maceration only gave you one. As of a few versions back, All raw ores macerate into 2 crushed unless it's an ore that yields more naturally via a multiplier. Combined with the overturned nature of veins getting thousands of items per vein as it is. Is already really generous.

In short, this change is not as simple to answer as 'just add a config for it' as it has sweeping implications for each option to the entire mod. So for the foreseeable future. It will stay this way as it already caused headaches to mod balance in the past. And we are not in the position to handle this currently. Might be reconsidered in the future. But for now I'll close this issue and label it for future references. If we plan to do anything regarding oremultipliers/balance/fortune behaviors, I'll reopen this issue and elevate it to an active PR.
